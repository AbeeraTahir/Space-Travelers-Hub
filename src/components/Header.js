import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Header.css';
import logo from '../images/logo.jpg';

const Header = () => (
  <header>
    <NavLink to="/">
      <div className="logo">
        <img src={logo} alt="logo" />
        <h1>Space Traverlers&apos; Hub</h1>
      </div>
    </NavLink>
    <nav>
      <ul>
        <li>
          ROCKETS
        </li>
        <li>
          MISSIONS
        </li>
        <li className="divider" />
        <li>
          MY PROFILE
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
