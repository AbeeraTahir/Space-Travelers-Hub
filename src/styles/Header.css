header {
  display: flex;
  padding: 10px 30px;
  align-items: center;
  justify-content: space-between;
}

.logo {
  display: flex;
  align-items: center;
  gap: 15px;
}

.logo img {
  width: 80px;
  height: 80px;
}

nav {
  padding: 0 20px;
}

nav ul {
  display: flex;
  align-items: center;
  gap: 40px;
  list-style-type: none;
}

nav ul li {
  font-size: 1rem;
}

a {
  text-decoration: none;
  color: black;
  opacity: 0.8;
}

.activelink {
  color: black;
  opacity: 1;
}

.divider {
  background-color: rgba(0, 0, 0, 0.705);
  width: 2px;
  height: 20px;
  opacity: 0.8;
}
