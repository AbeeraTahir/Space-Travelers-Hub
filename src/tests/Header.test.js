import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Header from '../components/Header';

it('Header component should render correctly', () => {
  const header = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <Header />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(header).toMatchSnapshot();
});
