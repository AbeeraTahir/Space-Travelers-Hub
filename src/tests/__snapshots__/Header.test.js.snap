// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <div
            class="logo"
          >
            <img
              alt="logo"
              src="logo.jpg"
            />
            <h1>
              Space Travelers' Hub
            </h1>
          </div>
        </a>
        <nav>
          <ul>
            <li>
              <a
                aria-current="page"
                class="activelink"
                href="/"
              >
                ROCKETS
              </a>
            </li>
            <li>
              <a
                href="/missions"
              >
                MISSIONS
              </a>
            </li>
            <li
              class="divider"
            />
            <li>
              <a
                href="/myprofile"
              >
                MY PROFILE
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="logo.jpg"
          />
          <h1>
            Space Travelers' Hub
          </h1>
        </div>
      </a>
      <nav>
        <ul>
          <li>
            <a
              aria-current="page"
              class="activelink"
              href="/"
            >
              ROCKETS
            </a>
          </li>
          <li>
            <a
              href="/missions"
            >
              MISSIONS
            </a>
          </li>
          <li
            class="divider"
          />
          <li>
            <a
              href="/myprofile"
            >
              MY PROFILE
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
